internal/planner/max.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package planner
+
+import (
+	"math"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/core"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+)
+
+type maxNode struct {
+	documentIterator
+	docMapper
+
+	p    *Planner
+	plan planNode
+
+	isFloat           bool
+	virtualFieldIndex int
+	aggregateMapping  []mapper.AggregateTarget
+
+	execInfo maxExecInfo
+}
+
+type maxExecInfo struct {
+	// Total number of times maxNode was executed.
+	iterations uint64
+}
+
+func (p *Planner) Max(
+	field *mapper.Aggregate,
+	parent *mapper.Select,
+) (*maxNode, error) {
+	isFloat := false
+	for _, target := range field.AggregateTargets {
+		isTargetFloat, err := p.isValueFloat(parent, &target)
+		if err != nil {
+			return nil, err
+		}
+		// If one source property is a float, the result will be a float - no need to check the rest
```

Comment:
I'm not seeing which part is incorrect.

Reply:
Okay I see what you mean. I've refactored the logic so it actually returns the correct type now. 

Correct me if I'm wrong but I don't think we have any operations that would allow us to test mixed type aggregates.

Reply:
> Correct me if I'm wrong but I don't think we have any operations that would allow us to test mixed type aggregates.

No you are correct, I'm only commenting on the comment, the code/tests are all good IMO :)

> I've refactored the logic so it actually returns the correct type now.

Nice, thanks!